toast-tk-web-agent/src/main/java/io/toast/tk/agent/web/AgentServerImpl.java

```diff
@@ -61,13 +72,59 @@ public boolean register(String ApiKey) {
 		return false;
 	}
 
+	private void openAliveSocket(String apiKey) throws InterruptedException, ExecutionException {
+		final String socketUrl = (app.getConfig().getWebAppUrl() + "/api/agent/stream/"+apiKey)
+				.replace("https://","wss://")
+				.replace("http://", "ws://")
+				.replace("//", "/")
+				.replace(":/", "://");
+
+		wsSlientProvider.get().prepareGet(socketUrl).execute(new WebSocketUpgradeHandler.Builder().addWebSocketListener(
+				new WebSocketTextListener() {
+					@Override
+					public void onOpen(WebSocket webSocket) {
+					}
+
+					@Override
+					public void onClose(WebSocket webSocket) {
+
+					}
+
+					@Override
+					public void onError(Throwable throwable) {
+
+					}
+
+					@Override
+					public void onMessage(String s) {
+
+					}
+				}
+		).build()).get();
+	}
+
+
+	private HttpRequest buildRequest(String uri, String json, String apiKey) {
+		HttpRequest request = HttpRequest.Builder.create().uri(uri).json(json).withKey(apiKey).build();
+		if(Boolean.valueOf(app.getConfig().getProxyActivate()) == true){
+			String proxyPort = app.getConfig().getProxyPort();
+			int port = proxyPort == null || proxyPort.isEmpty() ? -1 : Integer.valueOf(proxyPort).intValue();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use "Integer.parseInt" for this string-to-int conversion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2130)